spi/src/main/java/io/quarkiverse/kafkastreamsprocessor/spi/KStreamsProcessorConfigGenerator.java

```diff
@@ -0,0 +1,116 @@
+import io.quarkiverse.kafkastreamsprocessor.spi.properties.RetryConfig;
+import io.quarkiverse.kafkastreamsprocessor.spi.properties.SinkConfig;
+import io.quarkiverse.kafkastreamsprocessor.spi.properties.SourceConfig;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Accessors;
+
+class KStreamsProcessorConfigGenerator {
+
+    private static final String SOURCE_PREFIX = "kafkastreamsprocessor.input.sources.";
+
+    private static final String SOURCE_SUFFIX = ".topics";
+
+    private static final String SINK_PREFIX = "kafkastreamsprocessor.output.sinks.";
+
+    private static final String SINK_SUFFIX = ".topic";
+
+    static KStreamsProcessorConfig buildConfig(Config config) {
+        InputConfigImpl input = new InputConfigImpl(config.getOptionalValue("kafkastreamsprocessor.input.topic", String.class),
+                config.getOptionalValues("kafkastreamsprocessor.input.topics", String.class));
+        fillSources(config, input);
+        OutputConfigImpl output = new OutputConfigImpl(
+                config.getOptionalValue("kafkastreamsprocessor.output.topic", String.class));
+        fillSinks(config, output);
+        return new KStreamsProcessorConfigImpl(input, output);
+    }
+
+    private static void fillSources(Config config, InputConfigImpl input) {
+        for (String property : config.getPropertyNames()) {
+            if (property.startsWith(SOURCE_PREFIX) && property.endsWith(SOURCE_SUFFIX)) {
+                String sourceName = property.substring(SOURCE_PREFIX.length(), property.length() - SOURCE_SUFFIX.length());
+                String topics = config.getValue(property, String.class);
+                SourceConfig sourceConfig = new SourceConfigImpl(List.of(topics.split(",")));
+                if (sourceName.contains(".")) {
+                    throw new IllegalStateException("Parsed source name has a dot: " + sourceName);
+                }
+                input.sources().put(sourceName, sourceConfig);
+            }
+        }
+    }
+
+    private static void fillSinks(Config config, OutputConfigImpl output) {
+        for (String property : config.getPropertyNames()) {
+            if (property.startsWith(SINK_PREFIX) && property.endsWith(SINK_SUFFIX)) {
+                String sinkName = property.substring(SINK_PREFIX.length(), property.length() - SINK_SUFFIX.length());
+                String topic = config.getValue(property, String.class);
+                SinkConfig sinkConfig = new SinkConfigImpl(topic);
+                if (sinkName.contains(".")) {
+                    throw new IllegalStateException("Parsed sink name has a dot: " + sinkName);
+                }
+                output.sinks().put(sinkName, sinkConfig);
+            }
+        }
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    @Accessors(fluent = true)
+    private static class KStreamsProcessorConfigImpl implements KStreamsProcessorConfig {
```

Comment:
I have the feeling that this module is no more a SPI module, since we provide the implementation here, why don't change the name to quarkus-kafka-streams-processor-config : that is a module that contains the logic to parse smallrye Config to our Kstream*Config items

Reply:
So actually this implementation is a "workaround" to still be able to build the mappings during the build time (in a Quarkus build time BuildStep method) using the pure smallrye config Config object.
And keep the KStreamsProcessorConfig ConfigMapping as runtime configuration.

The module's readme was enriched to showcase how this can be used.